### PR TITLE
add a code-lstlisting template

### PR DESCRIPTION
**Pull request by Cancelll**

### CODE REVIEW

File: template/code-lstlisting.tex [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+			}
+			void print_hi() {
+				printf("Hi, world!");
+			}
+		\end{lstlisting}
+		\caption{Example c}
+		\label{code:sample:c}
+	\end{subfigure}%
+	\hfill
+	%\hspace{0.04\textwidth}%
```
Comment (yqtianust) [inline template/code-lstlisting.tex:38]: delete this

File: template/code-lstlisting.tex [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	\begin{subfigure}{0.48\textwidth}
+		\begin{lstlisting}[numbers=left,xleftmargin=2em]
+			int main() {
+				printf("Hello world!");
+			}
+		\end{lstlisting}
+		\caption{Example d}
+		\label{code:sample:d}
+		%\vspace*{1pt}
```
Comment (yqtianust) [inline template/code-lstlisting.tex:47]: remove this

**Review by yqtianust — CHANGES_REQUESTED** [review]
please remove the comments. the other part looks good.

**Review by yqtianust — APPROVED** [review]

**Comment by yqtianust** [comment]
Looks good to me. I think we can merge it.

**Review by chengniansun — CHANGES_REQUESTED** [review]
Can we add some comments at the top of this file on the purpose of this template?

**Review by chengniansun — APPROVED** [review]

**Pull request merged**